src/js/cli.js

```diff
@@ -211,7 +215,12 @@ function getCLIOpts(): CLIOptsType {
 
   const optind = parser.optind();
   if (!foundMainOpt && optind < argc) {
-    ret.mainScript = argv[optind];
+    if (argv[optind] === '-') {
```

Comment:
making this a special case causes `repl` to be set to `true`. You shouldn't need to worry about going to the REPL. I suggest keeping `mainScript === '-'` and treating that in `cljs.js`.